### PR TITLE
:bug: Log error if GITHUB_AUTH_TOKEN is unset

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds log to inform user.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

**Review by naveensrinivasan — APPROVED** [review]

File: cmd/root.go [hunk 1]
```diff
@@ -67,6 +67,10 @@ or ./scorecard --{npm,pypi,rubgems}=<package_name> [--checks=check1,...] [--show
 	Short: "Security Scorecards",
 	Long:  "A program that shows security scorecard for an open source software.",
 	Run: func(cmd *cobra.Command, args []string) {
+		if token, exists := os.LookupEnv("GITHUB_AUTH_TOKEN"); !exists || token == "" {
+			log.Fatalf("GITHUB_AUTH_TOKEN env var is not set. Please set this to your Github PAT before running this command.")
```
Comment (oliverchang) [inline cmd/root.go:71]: Nice, maybe link to https://github.com/ossf/scorecard#authentication (or https://docs.github.com/en/free-pro-team@latest/developers/apps/about-apps#personal-access-tokens) as well?

**Review by oliverchang — COMMENTED** [review]
nice!

File: cmd/root.go (hunk 1 above)
Comment (azeemshaikh38) [reply cmd/root.go:71]: Yes, we should. @laurentsimon FYI.

**Comment by ig33** [comment]
Money

**Comment by ig33** [comment]
Help

**Pull request merged**